inst/webshot.js

```diff
@@ -23,60 +27,64 @@ var opt_defaults = {
+if (args.length < 2) {
+  console.log(
+    'usage: phantomjs webshot.js <optsList>\n' +
+    '\n' +
+    'optsList is a JSON array containing configuration for each screenshot.\n' +
+    'For instance:\n' +
+    '\'[{"url":"url1.html","file":"file1.png"},{"url":"url2.html","file":"fil2.png","zoom":2}]\'');
   phantom.exit(1);
 }
 
-var url = args[1];
-var filename = args[2];
-var opts = utils.parseArgs(args.slice(3));
-opts = utils.merge(opt_defaults, opts);
-
-// These should be numbers
-if (opts.vwidth)  opts.vwidth  = +opts.vwidth;
-if (opts.vheight) opts.vheight = +opts.vheight;
-if (opts.delay)   opts.delay   = +opts.delay;
-if (opts.zoom)    opts.zoom    = +opts.zoom;
-
-// This should be four numbers separated by ","
-if (opts.cliprect) {
-  opts.cliprect = opts.cliprect.split(",");
-  opts.cliprect = opts.cliprect.map(function(x) { return +x; });
-}
-
-// Can be 1 or 4 numbers separated by ","
-if (opts.expand) {
-  opts.expand = opts.expand.split(",");
-  opts.expand = opts.expand.map(function(x) { return +x; });
-  if (opts.expand.length !== 1 && opts.expand.length !== 4) {
-    console.log("'expand' must have either 1 or 4 values.");
-    phantom.exit(1);
-  }
-}
-
-// Can have multiple selectors
-if (opts.selector) {
-  opts.selector = opts.selector.split(",");
-}
+var optsList = JSON.parse(args[1]);
 
 // =====================================================================
 // Screenshot
 // =====================================================================
-casper.start(url).zoom(opts.zoom)
-  .viewport(opts.zoom * opts.vwidth, opts.zoom * opts.vheight);
 
-if (opts.delay > 0)
-  casper.wait(opts.delay * 1000);
+casper.start();
+casper.options.onLoadError = function(c, url) {
+  console.log("Can not load ", url);
+  phantom.exit(1);
+};
 
-if (opts.eval) {
-  eval(opts.eval);
-}
+casper.eachThen(optsList, function(response) {
+  var opts = response.data;
+
+  // Prepare options
+  opts = utils.merge(opts, opt_defaults);
 
-casper.then(function() {
-  var cr = findClipRect(opts, this);
-  this.capture(filename, cr);
+  // This should be four numbers separated by ","
+  if (opts.cliprect) {
+    opts.cliprect = opts.cliprect.split(",");
+    opts.cliprect = opts.cliprect.map(function(x) { return +x; });
+  }
+
+  // Can be 1 or 4 numbers separated by ","
+  if (opts.expand) {
+    opts.expand = opts.expand.split(",");
+    opts.expand = opts.expand.map(function(x) { return +x; });
+    if (opts.expand.length !== 1 && opts.expand.length !== 4) {
+      console.log("'expand' must have either 1 or 4 values.");
+      phantom.exit(1);
+    }
+  }
+
+  // Can have multiple selectors
+  if (opts.selector) {
+    opts.selector = opts.selector.split(",");
+  }
+
+  // Go to url and perform the desired screenshot
+  casper.zoom(opts.zoom)
+    .viewport(opts.zoom * opts.vwidth, opts.zoom * opts.vheight)
+    .thenOpen(opts.url, function(r) {
+      casper.wait(opts.delay * 1000, function() {
```

Comment:
I fill unconfortable with this convention but indeed it is the case. I will change it back :)